src/chains/ethereum/transaction/tests/index.test.ts

```diff
@@ -1,6 +1,317 @@
 import assert from "assert";
-import ethereumTransaction from "../";
+import {
+  EIP2930AccessListTransaction,
+  LegacyTransaction,
+  TransactionFactory,
+  TypedDatabaseTransaction,
+  TypedRpcTransaction
+} from "../../transaction";
+import Common from "@ethereumjs/common";
+import Wallet from "../../ethereum/src/wallet";
+import { EthereumOptionsConfig } from "../../options";
+import { BUFFER_EMPTY, Quantity } from "@ganache/utils";
 
-describe("@ganache/ethereum-transaction", () => {
-  it("needs tests");
+describe("@ganache/ethereum-transaction", async () => {
+  const common = Common.forCustomChain(
+    "mainnet",
+    {
+      name: "ganache",
+      chainId: 1337,
+      comment: "Local test network",
+      bootstrapNodes: []
+    },
+    "berlin"
+  );
+  const options = EthereumOptionsConfig.normalize({});
+  const wallet = new Wallet(options.wallet);
```

Comment:
You can get the actual private keys, instead of a generating a fake one via `wallet.initialAccounts`.

You can also initialize the options with `wallet: {accounts: [{secretKey: "0x" + "46".repeat(32), balance: 100n}]`. This could be a decent way testing deterministically, allowing us to hardcode results for things test that check the hash and signature fields on the transactions.

* the `"46".repeat(32)` secret key is an example key used in an EIP-155 (https://eips.ethereum.org/EIPS/eip-155, which is the only reason I suggested it.

Reply:
Updated tests to set private key.